src/components/sidebar/sidebar.module.css

```diff
@@ -16,3 +16,41 @@
   margin: 0;
   padding: 0;
 }
+
+/*
+***
+* Classes for the mobile sidebar's level button
+***
+*/
+
+/* To left-align with the filter input */
+.levelButton {
+  margin-left: 1px;
+}
+
+/* To left-align the button icon with headings and nav item text */
+.levelButtonWithLeadingIcon {
+  /* TODO: need padding-right so padding looks even */
+  padding-left: 2px;
+}
+
+/* To left-align the button text with headings and nav item text */
+.levelButtonWithTrailingIcon {
+  /* TODO: need padding-right so padding looks even */
+  padding-left: 7px;
+}
+
+/* Spacing after the button when there is a filter input following */
+.levelButtonBeforeFilterInput {
+  margin-bottom: 6px;
+}
+
+/* Spacing after the button when there is a heading following  */
+.levelButtonBeforeHeading {
+  margin-bottom: 10px;
+}
+
+/* Spacing after the button when there is a nav item following */
+.levelButtonBeforeNavItem {
+  margin-bottom: 10px;
+}
```

Comment:
This feels bad 😬  Not necessarily a fault of the implementation, but it feels like the button is too aware of the context it's in.

Thinking out loud, I wonder if we could bake some of these spacing concerns into the other components? Have the filter input / heading / nav item components specify their vertical spacing needs and remove the need for level button to be aware of all of this.

Reply:
I don't see this as the button being aware of its context, but as `Sidebar` being aware since it's the component that applies these styles based on the props it's given. If the button component was in a different file than `Sidebar` and the styles were defined there, I would consider that being the button being too aware of its surroundings.

We run into the same problem if we try to apply a `margin-top` to elements like the headings. If the filter input comes before a heading, the spacing between them is `16px`. If the back-to link/button appears before a heading, the spacing between them is `20px`. Sometimes a heading is the first element in the `Sidebar`, and in that case it should have 0 margin above it because the `SidebarSidecarLayout` handles applying a consistent `padding-top: 32px` to the container.

Reply:
Right, so if each element had consistent spacing we could apply padding inside of each different element type so we wouldn't need to conditionally apply spacing based on the surrounding elements. E.g. a heading element might have more padding, whereas the filter input would have minimal padding so when either are vertically stacked the visual spacing between the elements would appear correct, without conditional spacing on the button.

Ultimately it's largely just different ways to accomplish the same thing, so I'm fine with either approach. Just an observation here!

Reply:
Unfortunately the spacing isn't consistent in a way that we could apply the styles without conditionals. 😕

Here are the possible `Sidebar` variations and how much space there should be between the elements, sharing so my thought process on this approach is clearer:

1. `StandaloneLink` <- `20px` -> `SidebarHeading` ([see here](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/?node-id=7502%3A69288))
2. `StandaloneLink` <- `20px` -> `SidebarNavMenuItem` ([see here](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/?node-id=7984%3A83910))
3. `StandaloneLink` <- `16px` -> `SidebarFilterInput` <- `16px` -> `SidebarHeading` ([see here](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/?node-id=7502%3A69204))
4. `SidebarHeading` (no space above) ([see here](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/?node-id=7502%3A69198))

My thought process:

- 4 tells me that `SidebarHeading` should not have any spacing set above it.
- That means to solve 1, `StandaloneLink` needs 20px spacing below it.
- That would also solve for 2, but then we have 3 where the spacing after `StandaloneLink` is smaller.
- To solve 3, alternatively, we could set `16px` top/bottom spacing on `SidebarFilterInput` and no spacing on `StandaloneLink`, but then that would break 1.

Reply:
Removed these classes, no longer relevant! Resolving this thread ✨